templates/java/org/prism/Loader.java.erb

```diff
@@ -79,16 +79,18 @@ public class Loader {
 
     private final ByteBuffer buffer;
     private final Nodes.Source source;
-    private String encodingName;
+    protected String encodingName;
+    <%- if string_type == "String" -%>
     private Charset encodingCharset;
+    <%- end -%>
```

Comment:
Does the extra field cause some warning or so? Otherwise it might be simpler to just leave it in even if unused for simplicity.
But no big deal, fine as you changed it here.

Probably the binary->StandardCharsets.US_ASCII mapping is quite specific to j.l.String, for RubySymbol you'd want to keep it binary. So then it's nice to remove the field for non-j.l.String to remove any incorrect usage :+1: